examples/blocks/MediaBlock.js

```diff
@@ -0,0 +1,102 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+import { Icon } from '../../lib';
+
+import Tooltip from '../components/Tooltip';
+import Portal from '../components/Portal';
+
+// Defined in both CSS and JS to constraint the maximum size of the tooltip.
+const OPTIONS_MAX_WIDTH = 300;
+const OPTIONS_SPACING = 70;
+const TOOLTIP_MAX_WIDTH = OPTIONS_MAX_WIDTH + OPTIONS_SPACING;
+
+const propTypes = {
+    blockProps: PropTypes.shape({
+        entity: PropTypes.object,
+        entityConfig: PropTypes.object.isRequired,
+    }).isRequired,
+    src: PropTypes.string.isRequired,
+    alt: PropTypes.string.isRequired,
+    children: PropTypes.node.isRequired,
+};
+
+/**
+ * Editor block to preview and edit images.
+ */
+class MediaBlock extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showTooltipAt: null,
+        };
+
+        this.openTooltip = this.openTooltip.bind(this);
+        this.closeTooltip = this.closeTooltip.bind(this);
+        this.renderTooltip = this.renderTooltip.bind(this);
+    }
+
+    openTooltip(e) {
+        const trigger = e.target;
+
+        this.setState({
+            showTooltipAt: Object.assign(trigger.getBoundingClientRect(), {
+                containerWidth: trigger.parentNode.offsetWidth,
+            }),
+        });
+    }
+
+    closeTooltip() {
+        this.setState({ showTooltipAt: null });
+    }
+
+    renderTooltip() {
+        const { children } = this.props;
+        const { showTooltipAt } = this.state;
+        const maxWidth = showTooltipAt.containerWidth - showTooltipAt.width;
+        const direction = maxWidth >= TOOLTIP_MAX_WIDTH ? 'left' : 'top-left';
```

Comment:
"Responsive" tooltip, but only when it is opened. The `resize` event handler isn't worth it.